ChaosMod/Effects/db/Player/GolfPercent.cpp

```diff
@@ -0,0 +1,98 @@
+/*
+	Effect by Vladimir Gotsin
+*/
+
+#include <stdafx.h>
+
+static void OnStart()
+{
+	//vehicle
+	auto playerPed = PLAYER_PED_ID();
+	float playerHeading = GET_ENTITY_HEADING(playerPed);
+	float heading = GET_ENTITY_HEADING(IS_PED_IN_ANY_VEHICLE(playerPed, false) ? GET_VEHICLE_PED_IS_IN(playerPed, false) : playerPed);
+
+	Vector3 playerPos = GET_ENTITY_COORDS(playerPed, false);
+
+	static const Hash caddyModel = GET_HASH_KEY("CADDY");
+
+	Vehicle veh = CREATE_VEHICLE(caddyModel, playerPos.x, playerPos.y, playerPos.z, heading, true, true, true);
+
+	//removing Vehicle
+	Vehicle toRemove = GET_VEHICLE_PED_IS_IN(playerPed, false);
+	SET_ENTITY_AS_MISSION_ENTITY(toRemove, true, true);
+	DELETE_VEHICLE(&toRemove);
```

Comment:
The current vehicle should not be removed, as it leads to softlocks if the vehicle is a mission entity